sys/hashes/sha256.c

```diff
@@ -318,3 +319,78 @@ const unsigned char *hmac_sha256(const unsigned char *key,
 
     return result;
 }
+
+unsigned char *sha256_chain(const unsigned char *seed, size_t seed_length,
+                            size_t elements,
+                            unsigned char *tail_element)
```

Comment:
was intentional, its my lousy try to sort the parameters semantically.
But I see its not evident, I'll remove it here and in the header.